src/itwinai/torch/distributed.py

```diff
@@ -56,6 +59,30 @@ def wrapper(self: "TorchDistributedStrategy", *args, **kwargs):
     return wrapper
 
 
+def _initialize_ray() -> None:
+    """This method is used by the RayDDPStrategy and RayDeepSpeedStrategy to initialize
+    the Ray backend if it is not already initialized.
+
+        Raises:
+            EnvironmentError: If required environment variables `HEAD_NODE_PORT` or
+                `HEAD_NODE_IP` are not set.
+                These should be set from the slurm script where the ray cluster is launched.
+    """
+    if ray.is_initialized():
+        return
+
+    HEAD_NODE_PORT = os.environ.get("HEAD_NODE_PORT")
+    HEAD_NODE_IP = os.environ.get("HEAD_NODE_IP")
```

Comment:
where are they used?

Reply:
Okay the official ray script exports 
```
port=6379
ip_head=$head_node_ip:$port
export ip_head
echo "IP Head: $ip_head"
```
but not 
`export head_node_ip="$head_node"I`
that is only in Eric's script. To be honest I don't know why he has it

Reply:
after a quick look online it seems to me that these env vars are useless in the python code. I would simply remove them and see if the code runs :)

Reply:
I'll push a new version where I have removed `export head_node_ip="$head_node"I` and renamed the variables to their original version. I agree that maybe the names are not the most explicit, but they are used consistently this way in all examples and in the documentation, so I feel like ensuring consistency here is more important, if we rename the variables it confuses things too much in my opinion 😌 But good that you pointed it out! I had just copied Eric's code but it could be simplified a bit apparently

Reply:
yes, I would remove all exports whatsoever. They don't seem to be needed anywhere

Reply:
Okay, this is done in the new version :)